Source/Core/Common/File.h

```diff
@@ -0,0 +1,233 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <cstdio>
+#include <ios>
+#include <istream>
+#include <iterator>
+#include <memory>
+#include <streambuf>
+#include <string>
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <dirent.h>
+#endif
+
+#include "Common/CommonTypes.h"
+#include "Common/NonCopyable.h"
+
+namespace File
+{
+class ReadOnlyFile
+{
+public:
+  virtual ~ReadOnlyFile() {}
```

Comment:
This should be defaulted and be in the cpp file, otherwise a compiler can generate out-of-line vtables.